File: c/account.c
#include<stdio.h>
#include<string.h>
struct account
{
	char *id;
	char *pwd;
     	int accno;
     	float bal;
};
main()
{
	struct account customer[3]= {{"amit","abc",123,1230.91}, {"ritesh","efg",456,19045.32},{"sajid","xyz",789,10234.57}};
   	int i=0,t=0,cnt=0;
	char *p,*uid;
	login:
	printf("Enter the UserID :");
	gets(uid);
	for(i=0;i<3;i++)
		if(!strcmp(uid,customer[i].id))
		{
			t=1;
			break;
		}
	if(t)
	{
		if(pwdCheck(customer[i]))
		{
			char r;
			printf("Login Succesful!\n");
			printf("Account Details Are :\n");
			printf("NAME = %s\nAccount No.= %d\nBalance = %.2f\n",customer[i].id,customer[i].accno,customer[i].bal);
			printf("Do you want to transfer amount?(y/n)\n");
			scanf("%c",&r);
			if(r=='y')
			{
				int acc,t=0,j;
				transfer :
				printf("Enter the account no.:");
				scanf("%d",&acc);
				for(j=0;j<3;j++)
					if(acc==customer[j].accno)
					{
						printf("NAME = %s\n",customer[j].id);
						t=1;
						break;
					}
				if(t)
				{
					float a;
					money:
					printf("Enter the amount :");
					scanf("%f",&a);
					if(a<=customer[i].bal)
					{
						customer[i].bal-=a;
						customer[j].bal+=a;
						printf("Transaction successful!\n");
						printf("Transaction Details are :\n");
					printf("transaction ID = 20192710\nAmount Transfered = %.2f\nBalance Left = %.2f\n",a,customer[i].bal);
					}
					else
					{
						printf("Insufficient balance!\n");
						printf("Your Balance is : %f\n",customer[i].bal);
						goto money;
					}
				}
				else
                {
					printf("Account no. is invalid\n");
					goto transfer;
                }
			}
			else
                printf("Logging Out!\n");
		}
		else
			printf("Login blocked!\n");
	}
	else
    {
		printf("Please Enter a Valid UserID\n");
		goto login;
    }
}
int pwdCheck(struct account customer)
{
	char *p;
	int t=0,cnt=0;
	attempt:
		printf("Enter the Password :");
		gets(p);
		if(!strcmp(p,customer.pwd))
			t=1;
		else
		{
			printf("Password is incorrect!\n");
			cnt++;
			if(cnt!=3)
			{
				printf("%d attempts left\n",3-cnt);
				goto attempt;
			}
		}
	return t;
}

File: python/important/account.py
import random as rd

l = [{"account no.": 123, "name": "Amit", "password": "abc", "balance": 13216.45},
     {"account no.": 456, "name": "Jack", "password": "efg", "balance": 1206.45},
     {"account no.": 789, "name": "Celina", "password": "ijk", "balance": 193036.45}]


def pwdCheck(pwd):
    i = 0
    while True:
        i += 1
        Pwd = input("Enter the password : ")
        if pwd != Pwd:
            if i < 3:
                print("Password is Incorrect\n{} attempts left!".format(3 - i))
                continue
            return False
        return True


def accOptions(d):
    a = int(input("Press 0 to Log Out\nPress 1 to Transfer Amount\nPress 2 to request Amount"))
    if a==1:
        acn = int(input("Enter Benificiary Ac/No. : "))
        for i in l:
            if i["account no."] == acn:
                print("Name = ", i["name"])
                bl = float(input("Enter Amount :"))
                if d["balance"] < bl:
                    print("Insufficient Balance")
                    accOptions(d)
                else:
                    d["balance"] -= bl
                    i["balance"] += bl
                    print("Transaction Successful")
                    print("Transaction ID =  ", rd.randint(100000, 1000000))
                    print("Balance left = ", d["balance"])
                break
        else:
            print("Account number does not exist!")
            accOptions(d)
    if a==2:
        acn = int(input("Enter Ac/No. : "))
        for i in l:
            if i["account no."] == acn:
                print("Name = ", i["name"])
                bl = float(input("Enter Amount :"))
                if i["balance"] < bl:
                    print("Insufficient Balance")
                    accOptions(d)
                else:
                    if pwdCheck(i["password"]):
                        i["balance"] -= bl
                        d["balance"] += bl
                        print("Transaction Successful")
                        print("Transaction ID =  ", rd.randint(1000000, 10000000))
                        print("Balance  = ", d["balance"])
                    break
        else:
            print("Account number does not exist!")
            accOptions(d)
    print("Successfully Log out")


def accDetails(i):
    print("Account Details are : ")
    for x, y in i.items():
        if x == "password":
            continue
        print(x, " = ", y)


def process():
    c = int(input("Press 0 to Create Account\nPress 1 to Login\n"))
    if c:
        uid = input("Enter User ID : ")
        for i in l:
            if i["name"] == uid:
                if pwdCheck(i["password"]):
                    print("Login Successful")
                    accDetails(i)
                    accOptions(i)
                    break
                print("Login Blocked")
                break
        else:
            print("User ID does not exist!")
            process()
    else:
        l.append({"account no.": rd.randint(100, 999), "name": input("Enter Name :"), "password": input("Enter Password :"),
              "balance": 0})
        print("Your Account  Created Successfully")
        for z in l:
            print(z)
        accDetails(l[-1])
        process()


process()
